### DISCUSSION TITLE
Finding first and last day for each grid point that meets a condition

### DISCUSSION THREAD

**Discussion by axelschweiger** [pydata/xarray#7656] [Q&A] [answered]
Hi I'm trying to find the first and last day of year that each grid point has a value (in this case ice concentration) below a threshold
This loop does what I need to do but it is very very  slow and probably a pretty dumb way to implement it. 

If tried a number of ways using where and .idxmin() but I don't  seem to be able to make this work 
Trying this operation on the entire array doesn't work

`FirstDoy=test.where(dsm <50,drop=True).time[0].dt.dayofyear`

Because drop can't reduce the dimension because it varies by grid point. The time coordinate is in cftime.DatetimeNoLeap 

There are some similar questions out there but the answers seem to be not correct. 

I suppose this could be a apply_ufunc but I haven't been able to figure that one out either. Array is 365,448,304. 

Seems like this shouldn't be that hard. Thanks for any help. 
```
import xarray as xr
import cftime
import datetime
import numpy as np

def find_FirstLast(ds):
    thresh=50.
    ntime,ny,nx=ds.shape   
    firstDOY=np.full((ny,nx),np.nan)
    lastDOY=np.full((ny,nx),np.nan)
    nday=np.full((ny,nx),np.nan)
    print(nx,ny)
    for i in range(0,ny): 
        print(i)
        for  j in range(0,nx):
#            print(i)
            temp=ds[:,i,j]
            temp = temp.where(temp < thresh,drop=True)
            if len(temp)> 0: 
                firstDOY[i,j]= temp[0].time.dt.dayofyear
                lastDOY[i,j]= temp[-1].time.dt.dayofyear
                nday[i,j]=lastDOY[i,j]-firstDOY[i,j]            
    return firstDOY,lastDOY,nday

ds=xr.open_dataset('check_minmax.nc')
dsm=ds.siconc
dsm.load()
print("Finished Loading Dataset")        
firstDOY,lastDOY,nday =find_FirstLast(dsm)

```

**Comment by spencerkclark** [comment, marked as answer]
Would the following work in your case?  It would need some adjustment to work in the scenario of multiple years, but it looks like your dataset only spans one year:

```python
ds = xr.open_dataset("check_minmax.nc")
threshold = 50.0
masked_dayofyear = ds.time.dt.dayofyear.where(ds.siconc < threshold)
first = masked_dayofyear.min("time")
last = masked_dayofyear.max("time")
days = last - first
```

**Reply by axelschweiger** [reply]
Thanks Spencer. This works perfectly and is very fast... and I'm feeling a bit "duh" why I didn't I figure this out... but well.

**Reply by spencerkclark** [reply]
No worries!
